ml-algorithms/src/test/java/org/opensearch/ml/engine/httpclient/MLHttpClientFactoryTests.java

```diff
@@ -29,84 +35,115 @@ public void test_getSdkAsyncHttpClient_success() {
+        } catch (IllegalArgumentException e) {
+            assertNotNull(e);
+        }
+
+        try {
+            MLHttpClientFactory.validate(HTTP, "3232300545", 80, PRIVATE_IP_DISABLED);
         } catch (IllegalArgumentException e) {
             assertNotNull(e);
         }
 
         try {
-            MLHttpClientFactory.validate("http", "192.11010049", 80, privateIpEnabled);
+            MLHttpClientFactory.validate(HTTP, "0:0:0:0:0:ffff:127.0.0.1", 80, PRIVATE_IP_DISABLED);
         } catch (IllegalArgumentException e) {
             assertNotNull(e);
         }
 
         try {
-            MLHttpClientFactory.validate("http", "3232300545", 80, privateIpEnabled);
+            MLHttpClientFactory.validate(HTTP, "153.24.76.232", 80, PRIVATE_IP_DISABLED);
         } catch (IllegalArgumentException e) {
             assertNotNull(e);
         }
 
         try {
-            MLHttpClientFactory.validate("http", "0:0:0:0:0:ffff:127.0.0.1", 80, privateIpEnabled);
+            MLHttpClientFactory.validate(HTTP, "177.0.0.1", 80, PRIVATE_IP_DISABLED);
         } catch (IllegalArgumentException e) {
             assertNotNull(e);
         }
 
         try {
-            MLHttpClientFactory.validate("http", "153.24.76.232", 80, privateIpEnabled);
+            MLHttpClientFactory.validate(HTTP, "12.16.2.3", 80, PRIVATE_IP_DISABLED);
         } catch (IllegalArgumentException e) {
             assertNotNull(e);
         }
     }
 
     @Test
     public void test_validateIp_rarePrivateIp_NotThrowException() throws Exception {
-        AtomicBoolean privateIpEnabled = new AtomicBoolean(true);
-        MLHttpClientFactory.validate("http", "0254.020.00.01", 80, privateIpEnabled);
-        MLHttpClientFactory.validate("http", "172.1048577", 80, privateIpEnabled);
-        MLHttpClientFactory.validate("http", "2886729729", 80, privateIpEnabled);
-        MLHttpClientFactory.validate("http", "192.11010049", 80, privateIpEnabled);
-        MLHttpClientFactory.validate("http", "3232300545", 80, privateIpEnabled);
-        MLHttpClientFactory.validate("http", "0:0:0:0:0:ffff:127.0.0.1", 80, privateIpEnabled);
-        MLHttpClientFactory.validate("http", "153.24.76.232", 80, privateIpEnabled);
+        MLHttpClientFactory.validate(HTTP, "0254.020.00.01", 80, PRIVATE_IP_ENABLED);
+        MLHttpClientFactory.validate(HTTP, "172.1048577", 80, PRIVATE_IP_ENABLED);
+        MLHttpClientFactory.validate(HTTP, "2886729729", 80, PRIVATE_IP_ENABLED);
+        MLHttpClientFactory.validate(HTTP, "192.11010049", 80, PRIVATE_IP_ENABLED);
+        MLHttpClientFactory.validate(HTTP, "3232300545", 80, PRIVATE_IP_ENABLED);
+        MLHttpClientFactory.validate(HTTP, "0:0:0:0:0:ffff:127.0.0.1", 80, PRIVATE_IP_ENABLED);
+        MLHttpClientFactory.validate(HTTP, "153.24.76.232", 80, PRIVATE_IP_ENABLED);
+        MLHttpClientFactory.validate(HTTP, "10.24.76.186", 80, PRIVATE_IP_ENABLED);
```

Comment:
should we add test for https too here?

Reply:
Only different logic for https is the resetting port number to 443 if user pass it as -1. Already there is a specific test case to handle this scenario.
Still I added 4 new test cases to cover valid and rare ip addresses of https.